utils/timeout.go

```diff
@@ -0,0 +1,18 @@
+package utils
+
+import (
+	"time"
+)
+
+// TimeoutChannel .
+func TimeoutChannel(timeout time.Duration) <-chan time.Time {
```

Comment:
y don't u employee the time.After or so instead of this function?

Reply:
ok